icu4c/source/common/uresbund.cpp

```diff
@@ -1031,8 +1031,12 @@ static UResourceBundle *init_resb_result(const ResourceData *rdata, Resource r,
                         /* this is an XPath alias, starting with "/LOCALE/" */
                         /* it contains the path to a resource which should be looked up */
                         /* starting in the requested locale */
-                        keyPath = locale; 
-                        locale = parent->fTopLevelData->fName; /* this is the requested locale's name */
+                        keyPath = locale;
+                        if (aliasBaseLocale != NULL) {
+                            locale = const_cast<char*>(aliasBaseLocale);
```

Comment:
Is this const_cast necessary?

(I see aliasBaseLocale is already a "const char*".)

Reply:
>I see aliasBaseLocale is already a "const char*".

Right.  I'm casting _away_ the const here.  Casting away a const is always kind of an iffy thing to do, but the original code seemed to be doing the moral equivalent of casting away the const (it's been a while, so I don't remember the details off the top of my head), and I couldn't think of a better approach.

>Is this const_cast necessary?

I was failing some of the automated PR checks without it.

Reply:
Is it possible to change the variable `locale` itself to be `const char*` ?

Reply:
I think the thing that prevents me from just making `locale` const is that we're writing through it on line 1026 above ("`*locale = 0;`").  I could split this into two variables, a const one to convey the locale to the `ures_openDirect()` call on line 1069 and another, temporary, non-const one just for the munging of the locale ID in ll. 1022-1028, but it felt like it'd be a lot simpler to just cast away the const and not worry about it.

I'll admit my thinking here might be influenced by the fact that I found this after already getting it to work and didn't want to redo the whole thing because of a compile error, but it still seems like it's not really worth the trouble.

Reply:
Aha: thanks a bunch for the explanation! Now it looks obvious. Some things only become obvious once I've bumped my head against them (or learned the language in a more thorough way).

I did not get very far in my review, but it looks like @sffc already took care of the parts where I could have been useful. :-) Bigger-picture appraisal of this work: thanks for this! I had struggled coming to terms with the existing inconsistencies between Java and C/C++.

Reply:
>Aha: thanks a bunch for the explanation! Now it looks obvious. Some things only become obvious once I've bumped my head against them (or learned the language in a more thorough way).

I understand.  It took me a while to go back through the code and figure that out again for myself!

>I did not get very far in my review, but it looks like @sffc already took care of the parts where I could have been useful. :-) Bigger-picture appraisal of this work: thanks for this! I had struggled coming to terms with the existing inconsistencies between Java and C/C++.

Thank you for the kind words and for taking the time to look!